### PR TITLE
Return the files and their digests for the object

### LINKED ISSUE #811: Return filenames / fixities for the API that DSA calls
So that we don't have to re-write the cocina when the cocina versions used by DSA and persisted in /purl don't match.

The key of the response will be the digest hash (use md5 for now) because there can be multiple versions with the same file name.  Also make it clear that it is a MD5 digest, in case we decide to change this later.

### PR DESCRIPTION
**Pull request by jcoyne**
This will allow DSA to query which files we already have

Fixes #811

### CODE REVIEW

File: app/models/public_json.rb [hunk 1]
```diff
@@ -8,4 +8,19 @@ def data=(data)
...
     ActiveSupport::Gzip.decompress(super)
   end
+
+  def cocina_hash
+    data && JSON.parse(data)
+  end
+
+  # The purpose here is to tell DSA which files we currently have (across all versions)
+  def files_by_md5
+    file_sets = cocina_hash.dig('structural', 'contains')
```
Comment (mjgiarlo) [inline app/models/public_json.rb:18]: It's not clear what this method returns if the `JSON.parse(data)` branch is not reached due to `data` being false-y. But in that case, does the thing returned respond to `#dig` (line 18)?

Comment (jcoyne) [reply app/models/public_json.rb:18]: I don't think this ever happens:
```
Purl.where(public_json: nil).count
 => 0
 ```

Comment (mjgiarlo) [reply app/models/public_json.rb:18]: If it's never `nil`, do we need the `data && ` part on line 13? Or are you saying that when that is false-y, `#files_by_md5` is never invoked?

Comment (jcoyne) [reply app/models/public_json.rb]: We no longer have that thanks to #816

Comment (mjgiarlo) [reply app/models/public_json.rb]: Excellent 🍻

**Pull request merged**